cms/search/tests/test_api_views.py

```diff
@@ -0,0 +1,132 @@
+        """Endpoint should return 404 when IS_EXTERNAL_ENV=False."""
+        response = self.client.get(RESOURCE_ENDPOINT)
+        self.assertEqual(response.status_code, 404)
+
+
+@override_settings(IS_EXTERNAL_ENV=False)
+class ResourceListViewPaginationTests(TestCase, ExternalAPITestMixin):
+    @classmethod
+    def setUpTestData(cls):
+        cls.index_page = IndexPageFactory(slug="custom-slug-1")
+        # One parent + 12 child pages = 13 total
+        cls.pages = [InformationPageFactory(slug=f"Page-{i}", parent=cls.index_page) for i in range(12)]
+        cls.total_resources = len(cls.pages) + 1  # +1 for the parent
+
+    def test_default_pagination_returns_first_slice(self):
+        """With no limit/ offset specified we should get DEFAULT_LIMIT (or all if fewer)."""
+        response = self.call_view_as_external("/v1/resources/")
+        self.assertEqual(response.status_code, 200)
+
+        data = self.parse_json(response)
+
+        expected_items = min(self.total_resources, 20)  # Default page size is 20
+        self.assertEqual(len(data["items"]), expected_items)
+        self.assertEqual(data["count"], expected_items)
+        self.assertEqual(data["total_count"], self.total_resources)
+        self.assertEqual(data["offset"], 0)
+
+    def test_second_slice_returns_remaining_items(self):
+        """Requesting offset=<7> should return whatever is left after the first slice."""
+        response = self.call_view_as_external(f"/v1/resources/?offset={7}")
+        self.assertEqual(response.status_code, 200)
+
+        data = self.parse_json(response)
+        expected_remaining = self.total_resources - 7
+        self.assertEqual(len(data["items"]), expected_remaining)
+        self.assertEqual(data["count"], expected_remaining)
+        self.assertEqual(data["offset"], 7)
+
+    def test_custom_limit_returns_requested_number(self):
+        response = self.call_view_as_external("/v1/resources/?limit=5")
+        self.assertEqual(response.status_code, 200)
+
+        data = self.parse_json(response)
+        self.assertEqual(len(data["items"]), 5)
+        self.assertEqual(data["count"], 5)
+        self.assertEqual(data["limit"], 5)
+
+    def test_limit_exceeds_max_uses_max(self):
+        """When limit exceeds max_limit, results should be capped at max_limit.
+        We add enough extra pages to go past MAX_LIMIT of 500.
+        """
+        extra_needed = 500  # over the cap
+        for i in range(extra_needed):
+            InformationPageFactory(slug=f"Extra-Page-{i}", parent=self.index_page)
+
+        response = self.call_view_as_external(f"/v1/resources/?limit={500 + 999}")
+        self.assertEqual(response.status_code, 200)
+
+        data = self.parse_json(response)
+        self.assertEqual(len(data["items"]), 500)
+        self.assertEqual(data["count"], 500)
+        self.assertEqual(data["limit"], 500)
+        self.assertGreater(data["total_count"], 500)
```

Comment:
please override the max limit setting rather than create so many pages.

but if you need to creat things in bulk, `InformationPageFactory.create_batch(number)` ([ref](https://factoryboy.readthedocs.io/en/stable/reference.html#factory.Factory.create_batch))

Reply:
Done!

I had to patch it, due to CustomLimitOffsetPagination.max_limit is fixed when pagination.py is first imported, and the `@override_settings` decorator by itself doesn't seem to override it. I was going to either reload the `pagination` module after the decorator has set the new value in settings, or `Monkey-patch` the frozen constant/ attribute.

Went with patching!